tfsdk/data_source_validation.go

```diff
@@ -0,0 +1,33 @@
+package tfsdk
+
+import (
+	"context"
+)
+
+// DataSourceConfigValidator describes reusable DataSource configuration validation functionality.
+type DataSourceConfigValidator interface {
```

Comment:
I'm certainly amenable to updating the name, I'm just not sure if there's any concerns with keeping the names and types relatively close to the RPCs similar to as has been done elsewhere. Should there be any future other implementations with differing semantics, it could be weird to have `DataSourceValidator` vs `DataSourceStateValidator` for example.